### DISCUSSION TITLE
costumtkinter showing plot by other file

### DISCUSSION THREAD

**Discussion by mhd-fauzan** [TomSchimansky/CustomTkinter#1596] [Announcements]
I've created a button to display a graphic from viewer.py file. I want the graphic from the viewer.py file to be displayed by clicking a button from the gui that I have created. Here's the code for a gui and graphics. Both are in separate files.
GUI.py Code
```
class Myapp():
    
    def __init__(self):
        self.root = customtkinter.CTk()
        self.root.geometry('1050x600')
        self.root.title("APx Platform")
        self.m1 = customtkinter.CTkButton(self.root, text="View Plot", font=("Ubuntu", 12))
        self.m1.grid(row=1, column=1, padx=(65, 65), pady=(5, 10))
app = Myapp()
app.root.mainloop()
```
And here is the viewer.py code
```
import pandas as pd
import matplotlib.pyplot as plt


class csv2df():
    
    def __init__(self):
        self.df = pd.read_csv("RMS level.csv", skiprows=[0,1,2])
        
    def plot(self):
        self.x = self.df["Hz"]
        self.y = self.df["dBSPL"]
        plt.plot(self.x, self.y)
        plt.xlabel("Frequency (Hz)")
        plt.ylabel("RMS Level (dBSPL)")
        
        plt.show()
        
data = csv2df()
data.plot()
```
Can you please fix it as I need this for a project.
Thank You.

I want to display the graph by clicking the "View Plot" button.

**Comment by helloHackYnow** [comment]
Are they in separate folder ?
If they aren't, you can probably just import your csv2df class like this : 
>from name_of_the_script import csvdf

From here, you can add : self.data = csv2df() in your init function.
As you have already initialize the csv2df, the command for the button can just be : command=self.data.plot

**Comment by Omena0** [comment]
Costumtkinter 😂
